sympy/polys/rings.py

```diff
@@ -3005,3 +3032,224 @@ def sqf_list(f, all=False):
+    ([1], None)
+
+    """
+    all_coeffs = polynomials
+
+    while True:
+        # Quick exits are most efficient if we start from the simplest polys
+        polynomials = sorted(set(all_coeffs), key=len)
+
+        # Find the intersection of symbols for each poly:
+        common = polynomials[0].main_variable()
+        allsame = True
+        for pi in polynomials[1:]:
+
+            # Quick exit
+            if not common:
+                R = polynomials[0].ring
+                K = R.domain
+                gcd = gcd_terms(polynomials, R, K)
+                return [gcd], None
+
+            syms = pi.main_variable()
+            if allsame and syms != common:
+                allsame = False
+            common &= syms
+
+        # The loop is complete if they all have the same symbols.
+        if allsame:
+            return polynomials, common
+
+        # Extract coefficients as polys containing only the common symbols.
+        all_coeffs = []
+        for i, pi in enumerate(polynomials):
+            coeffs_i = pi.coeff_wrt(pi.main_variable - common)
+            all_coeffs.extend(coeffs_i)
+
+            # Quick exit:
+            if len(coeffs_i) == 1:
+                R = polynomials[0].ring
+                K = R.domain
+                gcd = gcd_terms((all_coeffs + polynomials[i+1:]), R, K)
+                return [gcd], None
+
+def gcd_prs(polynomials):
+    """
+    Returns the greatest common divisor (GCD) of a set of polynomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import gcd_prs
+    >>> from sympy import ZZ, ring
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> polynomials = x - y, x - y
+    >>> gcd_prs(polynomials)
+    x - y
+
+    """
+    ring = polynomials[0].ring
+    domain = ring.domain
+
+    if any(len(pi) == 1 for pi in polynomials):
+        return gcd_terms(polynomials, ring, domain)
+
+    p, monom_gcd = monomial_extract(polynomials)
+
+    p, common_symbols = gcd_coeffs(polynomials)
+
+    gcd = p[0]
+    for pi in p[1:]:
+        gcd = _gcd_prs(gcd, pi)
+
+    if monom_gcd is not None:
+        gcd = gcd * monom_gcd
+
+    return gcd
+
+def _gcd_prs(p1, p2):
+    """
+    Returns the greatest common divisor (GCD) of two polynomials using the
+    Polynomial Resultant Sequences (PRS) method.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import _gcd_prs
+    >>> from sympy import ZZ, ring
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> f = 4*x**2 + 8*x + 10
+    >>> g = 2*x**2 + 6*x + 10
+    >>> _gcd_prs(f, g)
+    2
+
+    """
+    x = p1.main_variable()
+
+    c1, pp1 = p1.primitive()
+    c2, pp2 = p2.primitive()
+    from sympy.polys.domains import ZZ
+    R, _ = ring('_', ZZ)
+    c1, c2 = R(c1), R(c2)
+
+    h = pp1.subresultants(pp2, x)[-1]
+    c = gcd_prs([c1, c2])
+
+
+    domain = p1.ring.to_domain()
+
+    coeff = h.coeff_wrt(x, h.degree(x))
+
+    normalized_coeff = domain.canonical_unit(coeff)
+
+    if normalized_coeff == coeff:
+        h = -h
```

Comment:
This is still not right: you are assuming that it is just a question of a minus sign. The whole point of the `canonical_unit` function is that it might return something like `I` or `-I` and not just `1` or `-1`.